### PR TITLE
[MI-2776]:Fixed issue #912 on jira plugin

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-jira/issues/912

### CODE REVIEW

File: server/client_cloud.go [hunk 1]
```diff
@@ -114,3 +113,16 @@ func (client jiraCloudClient) ListProjects(query string, limit int) (jira.Projec
...
 }
+
+func (client jiraCloudClient) GetIssueTypes(projectID string) ([]*jira.IssueType, error) {
+	var result []*jira.IssueType
+	opts := map[string]string{
+		"projectId": projectID,
+	}
+
+	if err := client.RESTGet("3/issuetype/project", opts, &result); err != nil {
+		return result, err
```
Comment (ayusht2810) [inline server/client_cloud.go:124]:
```suggestion
		return nil, err
```

File: server/client_server.go [hunk 2]
```diff
@@ -175,7 +175,7 @@ func (client jiraServerClient) GetUserGroups(connection *Connection) ([]*jira.Us
 }
 
 func (client jiraServerClient) ListProjects(query string, limit int) (jira.ProjectList, error) {
-	plist, resp, err := client.Jira.Project.GetList()
+	plist, resp, err := client.Jira.Project.ListWithOptions(&jira.GetQueryOptions{})
```
Comment (ayusht2810) [inline server/client_server.go:178]:
```suggestion
	pList, resp, err := client.Jira.Project.ListWithOptions(&jira.GetQueryOptions{})
```

File: server/client_server.go [hunk 3]
```diff
@@ -188,3 +188,20 @@ func (client jiraServerClient) ListProjects(query string, limit int) (jira.Proje
...
+		IssueTypes []*jira.IssueType `json:"issueTypes"`
+	}
+
+	var result searchResult
+	opts := map[string]string{
+		"expand": "issueTypes",
+	}
+
+	if err := client.RESTGet(fmt.Sprintf("2/project/%s", projectID), opts, &result); err != nil {
+		return result.IssueTypes, err
```
Comment (ayusht2810) [inline server/client_server.go:203]: same

File: server/issue.go [hunk 4]
```diff
@@ -521,12 +521,18 @@ func (p *Plugin) httpGetJiraProjectMetadata(w http.ResponseWriter, r *http.Reque
 	projects := []utils.ReactSelectOption{}
 	issues := map[string][]utils.ReactSelectOption{}
 	for _, prj := range plist {
+		issueTypeList, err := p.GetIssueTypes(types.ID(instanceID), types.ID(mattermostUserID), prj.ID)
+		if err != nil {
+			p.API.LogError("Failed to Get issue types for project.", "Error", err, "ProjectKey", prj.Key)
```
Comment (ayusht2810) [inline server/issue.go:526]: I think we should first mention the project key and then the error

File: server/issue.go [hunk 5]
```diff
@@ -557,6 +563,18 @@ func (p *Plugin) ListJiraProjects(instanceID, mattermostUserID types.ID) (jira.P
 	return plist, connection, nil
 }
 
+func (p *Plugin) GetIssueTypes(instanceID, mattermostUserID types.ID, projectID string) ([]*jira.IssueType, error) {
+	client, _, _, err := p.getClient(instanceID, mattermostUserID)
+	if err != nil {
+		return nil, err
+	}
+	issueTypes, err := client.GetIssueTypes(projectID)
```
Comment (ayusht2810) [inline server/issue.go:571]:
```suggestion
	}
	
	issueTypes, err := client.GetIssueTypes(projectID)
```

File: server/issue.go [hunk 6]
```diff
@@ -557,6 +563,18 @@ func (p *Plugin) ListJiraProjects(instanceID, mattermostUserID types.ID) (jira.P
...
+	if err != nil {
+		return nil, err
+	}
+	return issueTypes, nil
```
Comment (ayusht2810) [inline server/issue.go:575]:
```suggestion
	}
	
	return issueTypes, nil
```

**Review by ayusht2810 — APPROVED** [review]

File: server/client_server.go [hunk 7]
```diff
@@ -175,16 +175,33 @@ func (client jiraServerClient) GetUserGroups(connection *Connection) ([]*jira.Us
...
-	result := *plist
+	result := *pList
 	if limit > 0 && len(result) > limit {
 		result = result[:limit]
 	}
 	return result, nil
 }
+
+func (client jiraServerClient) GetIssueTypes(projectID string) ([]*jira.IssueType, error) {
+	type searchResult struct {
```
Comment (manojmalik20) [inline server/client_server.go:193]: Shouldn't we create this at the top of this file?

File: server/issue.go [hunk 8]
```diff
@@ -521,12 +521,18 @@ func (p *Plugin) httpGetJiraProjectMetadata(w http.ResponseWriter, r *http.Reque
 	projects := []utils.ReactSelectOption{}
 	issues := map[string][]utils.ReactSelectOption{}
 	for _, prj := range plist {
+		issueTypeList, err := p.GetIssueTypes(types.ID(instanceID), types.ID(mattermostUserID), prj.ID)
+		if err != nil {
+			p.API.LogError("Failed to Get issue types for project.", "ProjectKey", prj.Key, "Error", err)
```
Comment (manojmalik20) [inline server/issue.go:526]:
```suggestion
			p.API.LogError("Failed to get issue types for project.", "ProjectKey", prj.Key, "Error", err.Error())
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**